### PR TITLE
add empty file check and tests for the same

### LINKED ISSUE #156: No check for empty file upload in common/src/file-upload.controller.ts
Currently in `common/src/file-upload.controller.ts` empty file uploads are allowed to be uploaded. Proposed PR adds a check before letting the upload take place.

This ticket helps with the following aspect
- Potential resource efficiency
- potential security gains (to prevent a mass empty file upload exploit)
- Only valid files pass through (improves data integrity)

### PR DESCRIPTION
**Pull request by gnaaruag**
## Checklist
Make sure you have
- [ ] Added relevant `@nestjs/swagger` decorators wherever required 
- [x] Added test cases (unit and integration) wherever required

## Description
- this adds a empty file upload check to the `file-upload.controller.ts` in `packages/common/src` 
- also adds tests for correct file upload, empty file upload
- if the file is found to be empty a status code of 400 is sent, else it is uploaded further according to `process.env.STORAGE_MODE`
- in `file-upload.interceptor` the line `request.raw = request.raw || request;` allows for valid request body to pass through

## Steps to test
**to test manually** 
create a post request to `/file/upload-file`  
test with empty file and non empty file via `multipart/form-data`

**to run automated test**
navigate to `samples/04-file-upload` and run `npm run test:e2e`

fixes #156

### CODE REVIEW

File: packages/common/test/app.e2e-spec.ts [hunk 1]
```diff
@@ -21,4 +23,41 @@ describe('AppController (e2e)', () => {
       .expect(200)
       .expect('Hello World!');
   });
+
+  it('/files/upload-file (POST); for file with content', async () => {
+    const testFilePath = path.join(__dirname, 'empty.txt');
+
+    fs.writeFileSync(testFilePath, 'abcd');
```
Comment (techsavvyash) [inline packages/common/test/app.e2e-spec.ts]: is this line required in the test case? doesn't seem to be getting used. If it is required, make sure to unlink the file as a test cleanup

File: packages/common/test/app.e2e-spec.ts [hunk 2]
```diff
@@ -21,4 +23,41 @@ describe('AppController (e2e)', () => {
...
+    const mockDestination = 'uploads';
+    const mockFilename = 'empty.txt';
+
+    const response = await request(app.getHttpServer())
+      .post(
+        `/files/upload-file?destination=${mockDestination}&filename=${mockFilename}`,
+      )
+      .attach('file', Buffer.from(''), mockFilename);
+
+    console.log('Response:', response.body);
```
Comment (techsavvyash) [inline packages/common/test/app.e2e-spec.ts]: remove the log statement.

File: sample/04-file-upload/test/app.e2e-spec.ts [hunk 3]
```diff
@@ -21,4 +21,38 @@ describe('AppController (e2e)', () => {
...
+    const mockDestination = 'uploads';
+    const mockFilename = 'empty.txt';
+
+    const response = await request(app.getHttpServer())
+      .post(
+        `/files/upload-file?destination=${mockDestination}&filename=${mockFilename}`,
+      )
+      .attach('file', Buffer.from(''), mockFilename);
+
+    console.log('Response:', response.body);
```
Comment (techsavvyash) [inline sample/04-file-upload/test/app.e2e-spec.ts]: remove the log

File: sample/04-file-upload/test/app.e2e-spec.ts
Comment (techsavvyash) [inline sample/04-file-upload/test/app.e2e-spec.ts:1]: we can remove these tests altogether I see no point in duplicating them across the app, if the `packages/common` passes then it's good enough

**Review by techsavvyash — CHANGES_REQUESTED** [review]

File: sample/04-file-upload/test/app.e2e-spec.ts
Comment (gnaaruag) [reply sample/04-file-upload/test/app.e2e-spec.ts:1]: we currently dont have any provision to run e2e tests in `packages/common` 
so we should let the tests run in `04-file-upload` and remove it from `packages/common`

Comment (gnaaruag) [reply sample/04-file-upload/test/app.e2e-spec.ts:1]: resolved
